docs/ref/models/fields.txt

```diff
@@ -831,6 +831,49 @@ exclusive. Any combination of these options will result in an error.
     instance, removing its time component. This is true for both storage and
     comparison.
 
+.. warning:: Arithmetic operations on :class:`DateField` may return a datetime on PostgreSQL and MySQL.
+
+    When performing arithmetic on a ``DateField`` with
+    :class:`~datetime.timedelta` on PostgreSQL and MySQL, the database
+    returns a ``datetime`` instead of a ``date``. This occurs because
+    Python's ``timedelta`` is converted to SQL ``INTERVAL``, and the SQL
+    operation ``date +/- interval`` returns a ``timestamp`` on these
+    databases.
+
+    To ensure a ``date`` result across all databases, use one of these
+    approaches:
+
+    Use :class:`~django.db.models.functions.Cast`
+    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Comment:
I generated the docs locally and I can't see this header nor the other one below.

This is one of the errors marked [in CI](https://github.com/django/django/actions/runs/19323334567/job/55268931679?pr=19947#step:5:15)

I think it's not possible to include subheaders inside admonitions, [per Sphinx](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#admonitions-messages-and-warnings).